packages/appstore/src/components/wallet-header/wallet-header.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import { Icon } from '@deriv/components';
+import classNames from 'classnames';
+import WalletCurrencyCard from './wallet-currency-card';
+import WalletHeaderButtons from './wallet-header-buttons';
+import WalletHeaderTitle from './wallet-header-title';
+import WalletHeaderBalance from './wallet-header-balance';
+import './wallet-header.scss';
+import { TAccountCategory, TAccountStatus, TWalletMaltaCurrency, TWalletSvgCurrency } from 'Types';
+import { getWalletHeaderButtons } from 'Constants/utils';
+
+type TWalletHeaderCommon = {
+    balance?: string;
+    is_open_wallet?: boolean;
+};
+
+type TWalletHeaderDemo = TWalletHeaderCommon & {
+    account_type: Extract<TAccountCategory, 'demo'>;
+    account_status?: never;
+    jurisdiction?: never;
+    currency?: never;
+};
+
+type TWalletHeaderSvg = TWalletHeaderCommon & {
+    account_status?: TAccountStatus;
+    account_type?: Extract<TAccountCategory, 'real'>;
+    jurisdiction: 'svg';
+    currency: TWalletSvgCurrency;
+};
+
+type TWalletHeaderMalta = TWalletHeaderCommon & {
+    account_status?: TAccountStatus;
+    account_type?: Extract<TAccountCategory, 'real'>;
+    jurisdiction: 'malta';
+    currency: TWalletMaltaCurrency;
+};
+
+type TWalletHeader = TWalletHeaderDemo | TWalletHeaderSvg | TWalletHeaderMalta;
+
+const WalletHeader = React.memo(
+    ({
+        account_status = '',
+        balance = '0.00',
+        currency = 'USD',
+        jurisdiction = 'svg',
+        account_type = 'real',
+        is_open_wallet = false,
+    }: TWalletHeader) => {
+        const [is_open, setIsOpen] = React.useState(is_open_wallet);
```

Comment:
```suggestion
        const [is_open, setIsOpen] = React.useState<boolean>(is_open_wallet);
```

Reply:
@thisyahlen-deriv I can add this type for useState, but in my opinion we don't need to do it with simple types (number, boolean, string and next), because TS is smart enough to define this type automatically from props type above

Reply:
@thisyahlen-deriv @mahdiyeh-deriv I have good example of using and not using generic for useState inside of deriv-app (image attached). I just want to know for the future how is right for our application. 
<img width="1057" alt="Screenshot 2023-05-02 at 11 19 19 AM" src="https://user-images.githubusercontent.com/120570511/235615708-3b142058-6ed7-4258-bc25-06806d051b78.png">

Reply:
@sergei you're right,
It is not mandatory to specify the type as TypeScript can infer the type from the  initial value but we need to make it consistent due to our code base. Let's do not add the types for simple and have  less code.